analyzers/dataframe/FCCAnalyses/ReconstructedParticleSource.h

```diff
@@ -0,0 +1,242 @@
+#ifndef ANALYZERS_SOURCE_RECONSTRUCTED_PARTICLE_H
+#define ANALYZERS_SOURCE_RECONSTRUCTED_PARTICLE_H
+
+// ROOT
+#include "ROOT/RVec.hxx"
+
+// EDM4hep
+#include "edm4hep/RecoMCParticleLinkCollection.h"
+#include "edm4hep/ReconstructedParticleCollection.h"
+
+namespace FCCAnalyses ::Source ::ReconstructedParticle {
+// --------------------  Selectors  -----------------------------------------
+
+/**
+ * \brief Analyzer to select reconstructed particles associated with MC
+ *        particle of the specified PDG ID.
+ */
+struct selPDG {
+  const int m_pdg;
+  /**
+   * \brief Constructor.
+   *
+   * \param[in] pdgID  Desired value of the PDG ID.
+   */
+  explicit selPDG(const int pdgID);
+  /**
+   * \brief Operator over the input link collection.
+   *
+   * \param[in] inLinkColl  Input collection of the reco-MC links.
+   * \param[out] result  Collection of the reconstructed particles associated
+   *                     with the MC particle of the desired PDG ID.
+   */
+  edm4hep::ReconstructedParticleCollection
+  operator()(const edm4hep::RecoMCParticleLinkCollection &inLinkColl);
+};
+
+/**
+ * \brief Analyzer to select reconstructed particles associated with MC
+ *        particle with the specified absolute value of the PDG ID.
+ */
+struct selAbsPDG {
+  const int m_absPdg;
+  /**
+   * \brief Constructor.
+   *
+   * \param[in] pdgID  Desired absolute value of the PDG ID.
+   */
+  explicit selAbsPDG(const int pdgID);
+  /**
+   * \brief Operator over the input link collection.
+   *
+   * \param[in] inLinkColl  Input collection of the MC-reco links.
+   * \param[out] result  Collection of the reconstructed particles associated
+   *                     with the MC particle with the desired absolute value
+   *                     of the PDG ID.
+   */
+  edm4hep::ReconstructedParticleCollection
+  operator()(const edm4hep::RecoMCParticleLinkCollection &inLinkColl);
+};
```

Comment:
The templated functions in `LinkSource` are named `sel_PDG`, but these here are `selPDG`. Is the naming difference done on purpose? I suppose there is pre-existing code with this difference as well, which is why it persists here?

Reply:
Good catch, the goal is to have camel case for all the source analyzers.